etc/spgw_u.conf

```diff
@@ -57,83 +57,114 @@ SPGW-U =
-                SCHED_POLICY = "SCHED_FIFO"; # Values in { SCHED_OTHER, SCHED_IDLE, SCHED_BATCH, SCHED_FIFO, SCHED_RR }
-                SCHED_PRIORITY = @THREAD_SGI_PRIO@;
-                POOL_SIZE = @SGI_THREADS@; # NUM THREADS
+                #CPU_ID        = 3;
+                SCHED_POLICY   = "SCHED_FIFO"; # Values in { SCHED_OTHER, SCHED_IDLE, SCHED_BATCH, SCHED_FIFO, SCHED_RR }
+                SCHED_PRIORITY = {{ env["THREAD_SGI_PRIO"] if "THREAD_SGI_PRIO" in env.keys() else '80' }};
+                POOL_SIZE      = {{ env["SGI_THREADS"] if "SGI_THREADS" in env.keys() else '8' }}; # NUM THREADS
             };
         };
     };
 
-    SNAT = "@NETWORK_UE_NAT_OPTION@"; # SNAT Values in {yes, no}
+    SNAT = "{{ env["NETWORK_UE_NAT_OPTION"] if "NETWORK_UE_NAT_OPTION" in env.keys() else 'no' }}"; # SNAT Values in {yes, no}
+
     PDN_NETWORK_LIST  = (
-                      {NETWORK_IPV4 = "@NETWORK_UE_IP@";} # 1 ITEM SUPPORTED ONLY
+                      {NETWORK_IPV4 = "{{ env["NETWORK_UE_IP"] }}";} # 1 ITEM SUPPORTED ONLY
                     );
 
     SPGW-C_LIST = (
-         {IPV4_ADDRESS="@SPGWC0_IP_ADDRESS@" ;}
+         {#- To be backward-compatible #}
+         {%- if env["SPGWC_HOSTNAME"] is defined %}
+         {IPV4_ADDRESS="{{ resolve(env["SPGWC_HOSTNAME"]) }}" ;}
+         {#- Force dummy value in 5G case #}
+         {%- elif env["ENABLE_5G_FEATURES"] is defined and env["ENABLE_5G_FEATURES"] == "yes" %}
+         {IPV4_ADDRESS="127.0.0.1" ;}
+         {%- else %}
+         {IPV4_ADDRESS="{{ env["SPGWC0_IP_ADDRESS"] }}" ;}
+         {%- endif %}
     );
 
     NON_STANDART_FEATURES :
     {
-        BYPASS_UL_PFCP_RULES = "@BYPASS_UL_PFCP_RULES@"; # 'no' for standard features, yes for enhancing UL throughput
+        BYPASS_UL_PFCP_RULES = "{{ env["BYPASS_UL_PFCP_RULES"] if "BYPASS_UL_PFCP_RULES" in env.keys() else 'no' }}"; # 'no' for standard features, yes for enhancing UL throughput
     };
 
-    SUPPORT_5G_FEATURES:  
+    SUPPORT_5G_FEATURES:
     {
+       {%- if env["ENABLE_5G_FEATURES"] is defined %}
        # STRING, {"yes", "no"},
-       ENABLE_5G_FEATURES = "@ENABLE_5G_FEATURES@" # Set to 'yes' to support 5G Features
-       REGISTER_NRF = "@REGISTER_NRF@";            # Set to 'yes' if UPF resgisters to an NRF
-       USE_FQDN_NRF = "@USE_FQDN_NRF@";            # Set to 'yes' if UPF relies on a DNS/FQDN service to resolve NRF's FQDN
-       UPF_FQDN_5G  = "@UPF_FQDN_5G@";             # Set FQDN of UPF 
+       ENABLE_5G_FEATURES = "{{ env["ENABLE_5G_FEATURES"] }}" # Set to 'yes' to support 5G Features
+       REGISTER_NRF = "{{ env["REGISTER_NRF"] }}";            # Set to 'yes' if UPF registers to an NRF
+       USE_FQDN_NRF = "{{ env["USE_FQDN_NRF"] if "USE_FQDN_NRF" in env.keys() else 'no' }}";            # Set to 'yes' if UPF relies on a DNS/FQDN service to resolve NRF's FQDN
+       UPF_FQDN_5G  = "{{ env["UPF_FQDN_5G"] if "UPF_FQDN_5G" in env.keys() else 'none' }}";             # Set FQDN of UPF
```

Comment:
@rdefosse I actually did localhost everywhere if not given. There is a problem to do localhost?

Reply:
@arora-sagar 

That's for me the bug in `NRF` --> if `USE_FQDN-NRF` is set to `no` (default), the `UPF_FQDN_5G` should be `garbage`.

But OK. I will put `localhost`